### PR TITLE
OP-143/sylius upgrade

### PR DESCRIPTION
**Pull request by gabriela-lubkowska**
- Added support for Sylius 1.12 
- Added webpack
- Removed swiftmailer
- Removed include in product variant tab details

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,18 +17,15 @@ jobs:
...
             fail-fast: false
             matrix:
-                php: [7.4, 8.0]
-                sylius: [~1.9.0, ~1.10.0, ~1.11.0]
+                php: [8.0]
+                sylius: [~1.11.0, ~1.12.0]
                 node: [14.x]
                 mysql: [5.7, 8.0]
 
                 exclude:
```
Comment (senghe) [inline .github/workflows/build.yml]: How about excluding Sylius 1.11 and Symfony 6.0?

File: composer.json [hunk 2]
```diff
@@ -34,16 +33,19 @@
...
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
+        "allow-plugins": {
+            "dealerdirect/phpcodesniffer-composer-installer": true,
+            "phpstan/extension-installer": true
+        }
     },
     "conflict": {
         "symfony/symfony": "4.1.8",
```
Comment (senghe) [inline composer.json]: Are the conflicts still current? We do not support Symfony 4.x now.

File: tests/Application/config/routes/sylius_shop.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ sylius_shop:
     resource: "@SyliusShopBundle/Resources/config/routing.yml"
     prefix: /{_locale}
     requirements:
-        _locale: ^[A-Za-z]{2,4}(_([A-Za-z]{4}|[0-9]{3}))?(_([A-Za-z]{2}|[0-9]{3}))?$
+        _locale: ^[a-z]{2}(?:_[A-Z]{2})?$
```
Comment (senghe) [inline tests/Application/config/routes/sylius_shop.yaml]: Why have you changed the regex? Does it have any connection to Sylius 1.12 upgrade?

File: tests/Application/config/sylius/1.11/bundles.php [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 return [
```
Comment (senghe) [inline tests/Application/config/sylius/1.11/bundles.php:9]: No OS header

File: tests/Application/config/sylius/1.12/bundles.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
```
Comment (senghe) [inline tests/Application/config/sylius/1.12/bundles.php:9]: No OS header

File: tests/Application/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "dependencies": {
     "babel-polyfill": "^6.26.0",
-    "chart.js": "^2.9.3",
+    "chart.js": "^3.7.1",
```
Comment (senghe) [inline tests/Application/package.json:4]: Have you checked the dashboard charts after bumping the library version?

File: tests/Application/package.json.~1.12.0.dist [hunk 7]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (senghe) [inline tests/Application/package.json.~1.12.0.dist]: The documentation update is necessary for the file-split.

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Application/package.json.~1.12.0.dist (hunk 7 above)
Comment (gabriela-lubkowska) [reply tests/Application/package.json.~1.12.0.dist]: Removed file split

File: tests/Application/config/sylius/1.12/bundles.php (hunk 5 above)
Comment (gabriela-lubkowska) [reply tests/Application/config/sylius/1.12/bundles.php:9]: Added in next commit

File: tests/Application/config/sylius/1.11/bundles.php (hunk 4 above)
Comment (gabriela-lubkowska) [reply tests/Application/config/sylius/1.11/bundles.php:9]: Added in next commit

File: tests/Application/config/routes/sylius_shop.yaml (hunk 3 above)
Comment (gabriela-lubkowska) [reply tests/Application/config/routes/sylius_shop.yaml]: Changed in next commit

File: composer.json (hunk 2 above)
Comment (gabriela-lubkowska) [reply composer.json]: Removed unnecessary conflicts

File: .github/workflows/build.yml (hunk 1 above)
Comment (gabriela-lubkowska) [reply .github/workflows/build.yml]: Added in next commit

**Review by senghe — APPROVED** [review]

File: tests/Application/package.json (hunk 6 above)
Comment (gabriela-lubkowska) [reply tests/Application/package.json:4]: Checked admin dashboard, looks fine in sylius 1.11 and 1.12. But needed to add weback config in next commit

**Review by senghe — APPROVED** [review]

**Pull request merged**